certora/specs/wadRayMath.spec

```diff
@@ -0,0 +1,96 @@
+methods {
+    wadMul(uint256, uint256)        returns (uint256)       envfree
+    wadDiv(uint256, uint256)        returns (uint256)       envfree
+    rayMul(uint256, uint256)        returns (uint256)       envfree
+    rayDiv(uint256, uint256)        returns (uint256)       envfree
+    rayToWad(uint256)               returns (uint256)       envfree
+    wadToRay(uint256)               returns (uint256)       envfree
+}
+
+definition WAD()        returns uint = 10^18;
+definition RAY()        returns uint = 10^27;
+definition WADTORAY()   returns uint = 10^9;
+
+/// wadMul ///
+
+rule wadMulSafety(uint256 a, uint256 b) {
+    uint res = wadMul(a, b);
+
+    assert res == (a * b + (WAD() / 2)) / WAD();
```

Comment:
It would be nice to also explicit the two cases. We could for example add the following assertion:
```solidity 
assert a * b % WAD() < WAD() / 2 => res == a * b / WAD();`

Reply:
You mean here and everywhere else ?

Reply:
I tried it, but I think that it is over complicating things